compiler-rt/lib/asan/asan_malloc_win_thunk.cpp

```diff
@@ -141,6 +141,24 @@ STATIC_MALLOC_INTERFACE void *_expand_dbg(void *, size_t, int, const char *,
   return nullptr;
 }
 
+// _aligned
+STATIC_MALLOC_INTERFACE void *_aligned_malloc(size_t size, size_t alignment) {
+  size = (size + alignment - 1) & ~(alignment - 1);
+  return __asan_malloc(size);
+}
+
+STATIC_MALLOC_INTERFACE void *_aligned_realloc(void *ptr, size_t size,
+                                            size_t alignment) {
+  size = (size + alignment - 1) & ~(alignment - 1);
+  return __asan_realloc(ptr, size);
+}
```

Comment:
This code doesn't actually guarantee the returned pointer will be aligned though.

I think the way to do this is to dllexport an __asan_aligned_{malloc,realloc} in asan_malloc_win.cpp, and dllimport and call them here -- look at __asan_malloc for an example.